Plugins/AdminGuiPlugin/Angular2ModelGenerator/Generators/Entities/DataStructureGenerator.cs

```diff
@@ -100,6 +103,31 @@ protected override string GenerateGetMethods(DataStructureInfo info)
             return new KeyValuePair<MenuItemType, AppMenuItem>(_itemType, null);
         }
 
+        protected override string GenerateIdFieldDefinition(DataStructureInfo info)
+        {
+            if (IsSimplerDataStructures(info))
+            {
+                return string.Empty;
+            }
+
+            return base.GenerateIdFieldDefinition(info);
+        }
+
+        protected override string GenerateIdFieldInitialization(DataStructureInfo info)
+        {
+            if (IsSimplerDataStructures(info))
+            {
+                return string.Empty;
+            }
+
+            return base.GenerateIdFieldInitialization(info);
+        }
+
+        private bool IsSimplerDataStructures(DataStructureInfo info)
+        {
+            return ConceptKeywords.SimplerDataStructures.Contains(info.GetType().GetCustomAttribute<ConceptKeywordAttribute>()?.Keyword);
```

Comment:
Checking for simple data structures by keyword does include for the derivations of these concepts (custom data structure concepts on our projects are often used and they are always derived from existing ones as far as I know). I think would be best to reverse the logic here and list the concepts with ID property, then check if the `info` is a derivation of such concept (probably with  Type.IsAssignableFrom method).

I have search the data structure concepts in the CommonConcepts, the concepts with ID that we should check here: PolymorphicInfo, ComputedInfo, QueryableExtensionInfo, SqlQueryableInfo, LegacyEntityWithAutoCreatedViewInfo, LegacyEntityInfo, BrowseDataStructureInfo, EntityInfo.